### PR TITLE
[chimera] Add new port

### PR DESCRIPTION
**Pull request by ZMZ91**
Chimera is a software regular expression matching engine that is a hybrid of Hyperscan and PCRE. 
https://github.com/intel/hyperscan/tree/master/chimera

### CODE REVIEW

**Comment by ZMZ91** [comment]
hi there, one question that how to handle the conflicts if hyperscan installed as well?

File: ports/chimera/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+message(STATUS "Source code copy done.")
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${CHIMERA_SOURCE_PATH}
+    PREFER_NINJA
+)
+
+message(STATUS "Cmake install starts ...")
+vcpkg_install_cmake()
+message(STATUS "Cmake install done.")
```
Comment (NancyLi1013) [inline ports/chimera/portfile.cmake]: `message() `seems to be not needed  here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/chimera/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH PCRE_SOURCE_PATH
+    ARCHIVE ${ARCHIVE_PCRE}
+    WORKING_DIRECTORY ${CHIMERA_SOURCE_PATH}
+)
+
+# rename the pcre source code folder to pcre-8.41 which is required by chimera
+file(RENAME "${PCRE_SOURCE_PATH}" "${CHIMERA_SOURCE_PATH}/pcre-${PCRE_VERSION}")
+message(STATUS "Source code copy done.")
+
```
[the remark this answers is not in the document]
Comment (ZMZ91) [reply ports/chimera/portfile.cmake]: Thanks for your review. I see your point that pcre is supplied by vcpkg. But here for chimera it requires pcre's source code at the location during building. If I don't download and copy pcre-8.41 there, the building will fail.

File: ports/chimera/portfile.cmake (hunk 1 above)
Comment (ZMZ91) [reply ports/chimera/portfile.cmake]: Thank you. Just resolved it.

File: ports/chimera/portfile.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (NancyLi1013) [reply ports/chimera/portfile.cmake]: Thanks for your info.

**Comment by NancyLi1013** [comment]
I noticed it failed on Windows platform.
Could you please help confirm if it supports Windows?

If not, please add `vcpkg_fail_port_install(ON_TARGET "Windows")` and also update the related triplets to ci.baseline.txt.
If yes, please look into the regressions and try to fix them.

Thanks.

File: ports/chimera/portfile.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZMZ91) [reply ports/chimera/portfile.cmake]: But find_package() or find_library() won't download the pcre code there, right? It's required by chimera that pcre source code there during building and build pcre stuff into chimera lib. Please checkout chimera's cmake as below
```cmake
# first look in pcre-$version or pcre subdirs
if (PCRE_SOURCE)
    # either provided on cmdline or we've seen it already
    set (PCRE_BUILD_SOURCE TRUE)
elseif (EXISTS ${PROJECT_SOURCE_DIR}/pcre-${PCRE_REQUIRED_VERSION})
    set (PCRE_SOURCE ${PROJECT_SOURCE_DIR}/pcre-${PCRE_REQUIRED_VERSION})
    set (PCRE_BUILD_SOURCE TRUE)
elseif (EXISTS ${PROJECT_SOURCE_DIR}/pcre)
    set (PCRE_SOURCE ${PROJECT_SOURCE_DIR}/pcre)
    set (PCRE_BUILD_SOURCE TRUE)
endif()

if (PCRE_BUILD_SOURCE)
    message(STATUS "SHOULD build pcre with source")
    if (NOT IS_ABSOLUTE ${PCRE_SOURCE})
        set(PCRE_SOURCE "${CMAKE_BINARY_DIR}/${PCRE_SOURCE}")
    endif ()
    set (saved_INCLUDES "${CMAKE_REQUIRED_INCLUDES}")
    set (CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES} ${PCRE_SOURCE}")

    if (PCRE_CHECKED)
        set(PCRE_INCLUDE_DIRS ${PCRE_SOURCE} ${PROJECT_BINARY_DIR}/pcre)
        set(PCRE_LDFLAGS -L"${LIBDIR}" -lpcre)

        # already processed this file and set up pcre building
        return()
    endif ()

    # first, check version number
    CHECK_C_SOURCE_COMPILES("#include <pcre.h.generic>
    #if PCRE_MAJOR != ${PCRE_REQUIRED_MAJOR_VERSION} || PCRE_MINOR < ${PCRE_REQUIRED_MINOR_VERSION}
    #error Incorrect pcre version
    #endif
    main() {}" CORRECT_PCRE_VERSION)
    set (CMAKE_REQUIRED_INCLUDES "${saved_INCLUDES}")

    if (NOT CORRECT_PCRE_VERSION)
        unset(CORRECT_PCRE_VERSION CACHE)
        message(STATUS "Incorrect version of pcre - version ${PCRE_REQUIRED_VERSION} or above is required")
        return ()
    else()
        message(STATUS "PCRE version ${PCRE_REQUIRED_VERSION} or above - building from source.")
    endif()

    # PCRE compile options
    option(PCRE_BUILD_PCRECPP OFF)
    option(PCRE_BUILD_PCREGREP OFF)
    option(PCRE_SHOW_REPORT OFF)
    set(PCRE_SUPPORT_UNICODE_PROPERTIES ON CACHE BOOL "Build pcre with unicode")
    message(STATUS "PCRE_SUPPORT_UNICODE_PROPERTIES : ${PCRE_SUPPORT_UNICODE_PROPERTIES}")
    add_subdirectory(${PCRE_SOURCE} ${PROJECT_BINARY_DIR}/pcre EXCLUDE_FROM_ALL)
    set(PCRE_INCLUDE_DIRS ${PCRE_SOURCE} ${PROJECT_BINARY_DIR}/pcre)
    set(PCRE_LDFLAGS -L"${LIBDIR}" -lpcre)
else ()
    # pkgconf should save us
    find_package(PkgConfig)
    pkg_check_modules(PCRE libpcre>=${PCRE_REQUIRED_VERSION})
    if (PCRE_FOUND)
        set(CORRECT_PCRE_VERSION TRUE)
        message(STATUS "PCRE version ${PCRE_REQUIRED_VERSION} or above")
    else ()
        message(STATUS "PCRE version ${PCRE_REQUIRED_VERSION} or above not found")
        return ()
    endif ()
endif (PCRE_BUILD_SOURCE)
```

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/chimera/portfile.cmake]: 1. set `PCRE_BUILD_SOURCE` to `false`.
2. modify `find_package(PkgConfig)` to `find_package(PCRE REQUIRED)`.
3. modify `pkg_check_modules(PCRE libpcre>=${PCRE_REQUIRED_VERSION})` to `link_libraries(${PCRE_LIBRARIES})`.
4. add [FindPCRE.cmake](https://github.com/LuaDist/pcre/blob/master/FindPCRE.cmake) to source.

[the remark this answers is not in the document]
Comment (ZMZ91) [reply ports/chimera/portfile.cmake]: Thanks @JackBoosY. So you suggest modifying chimera's cmake file from its repo? I'm not sure whether that's a good practice.

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/chimera/portfile.cmake]: That's reasonable,  I will give you suggestion.

[the remark this answers is not in the document]
Comment (ZMZ91) [reply ports/chimera/portfile.cmake]: Thank you @JackBoosY . I've created a pr for hyperscan/chimera for supporting building chimera without pcre source code. Will update this pr later.

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -234,6 +234,9 @@ chartdir:arm64-windows=fail
 chartdir:arm-uwp=fail
 chartdir:x64-uwp=fail
 chartdir:x64-windows-static=fail
+chimera:x64-windows=fail
+chimera:x64-windows-static=fail
+chimera:x86-windows=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt]: I noticed that you have added `vcpkg_fail_port_install(ON_TARGET "Windows")`.
But you only updated the above triplets here.
```
chimera:arm64-windows=fail
chimera:x64-uwp=fail
chimera:arm-uwp=fail
```
These triplets are also needed to add.

File: ports/chimera/CONTROL [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Source: chimera
+Version: 5.2.1
+Homepage: https://github.com/intel/hyperscan
+Description: Chimera is a software regular expression matching engine that is a hybrid of Hyperscan and PCRE. The design goals of Chimera are to fully support PCRE syntax as well as to take advantage of the high performance nature of Hyperscan.
+Build-Depends: boost-array, boost-chrono, boost-config, boost-core, boost-crc, boost-detail, boost-functional, boost-regex, boost-system, boost-thread, boost-type-traits, boost-unordered, boost-utility, boost-dynamic-bitset, boost-random, boost-graph, boost-multi-array, boost-icl, boost-ptr-container, pcre, python3, ragel
```
Comment (NancyLi1013) [inline ports/chimera/CONTROL]: Could you please add `Supports: linux|osx` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (ZMZ91) [reply scripts/ci.baseline.txt]: Thank you. Please check in the latest commit.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Hi @ZMZ91 

Let's wait for the PR https://github.com/intel/hyperscan/pull/223 merged.
Note: If `chimera `requires `pcre `to build, please use `pcre `that vcpkg provided.

Thanks.

**Comment by ZMZ91** [comment]
Hi @NancyLi1013, 
Got echo from https://github.com/intel/hyperscan/pull/223 which said `Chimera needs to use an internal UTF8 function _pcre_valid_utf from PCRE, which requires PCRE source to been compiled with --enable-unicode-properties option, so PCRE source is required.`
Referring the reply above, I'm afraid we can't cut pcre source from the installation.

**Comment by NancyLi1013** [comment]
Hi @ZMZ91 
Thanks for your above info.
We will have a internal discussion about this case.

**Comment by NancyLi1013** [comment]
@ZMZ91 
Sorry for the delay.

If we add `--enable-unicode-properties` option for `pcre `provided by vcpkg, can this port work?

**Comment by NancyLi1013** [comment]
@ZMZ91 
Is work still being done for this PR?

**Comment by ZMZ91** [comment]
Thanks @NancyLi1013 , sorry for delayed reply. But the option `--enable-unicode-properties` is not fixing it.

**Comment by JackBoosY** [comment]
I will handle this issue later.

**Comment by JackBoosY** [comment]
Do we need to add Hyperscan instead of chimera?

File: ports/chimera/CONTROL [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
```
Comment (JackBoosY) [inline ports/chimera/CONTROL:5]: According to [boost.cmake](https://github.com/intel/hyperscan/blob/master/cmake/boost.cmake#L33), this port should only depends on `boost-graph` and `boost-concept-check`.

**Comment by ZMZ91** [comment]
Actually this package is a super set of `hyperscan` which's already in vcpkg. `chimera` requires `hyperscan` and `pcre`.

**Comment by JackBoosY** [comment]
@ZMZ91 Can you resolve the file conflict first?

File: ports/chimera/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,49 @@
+vcpkg_fail_port_install(ON_TARGET "Windows")
+
+set(CHIMERA_VERSION "5.2.1")
+set(PCRE_VERSION "8.41")
+
+# download hyperscan source code which is required by and also contains chimera
+vcpkg_download_distfile(ARCHIVE_HYPERSCAN
+    URLS "https://github.com/intel/hyperscan/archive/d79973efb1fcf5ed338122882c1f896829767fb6.zip"
```
Comment (JackBoosY) [inline ports/chimera/portfile.cmake:8]: Since this repo is in github, can you replace it with `vcpkg_from_github`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
@ZMZ91 Since you already added `Supports` to _CONTROL_, we don't need to modify _ci.baseline.txt_.

**Comment by JackBoosY** [comment]
```
-- Downloading https://github.com/intel/hyperscan/archive/d79973efb1fcf5ed338122882c1f896829767fb6.tar.gz...
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:99 (message):
  

  File does not have expected hash:

          File path: [ /Users/vagrant/Data/downloads/temp/intel-hyperscan-d79973efb1fcf5ed338122882c1f896829767fb6.tar.gz ]
      Expected hash: [ 8bf6e8c323ab3c4b3f26d871a9f89e666edba598925c99ad2f14b3e849792c51cb87bc30ad54008ec813603a9a7c94d689ad4344e1a45c10b2a891e434a678f3 ]
        Actual hash: [ 00fe31805efddf25daceaecb94de44dae7cda8e5fb9593bb69ce2d3e3675b5c820cc6bac8d75f88f4ffe7dc9f3c4603b75816479eb36dba5bfb272dc6998003a ]
```

**Review by JackBoosY — APPROVED** [review]

**Comment by ZMZ91** [comment]
Hi there, anything needed from me before merging?

**Comment by JackBoosY** [comment]
Ping @ras0219 @ras0219-msft for merge this PR.

**Comment by NancyLi1013** [comment]
@ZMZ91 
We need to add `_pcre_valid_utf ` for `pcre `provided in vcpkg first in order to make it available in this PR.

**Comment by JackBoosY** [comment]
@ZMZ91 
Here is my changes based on this PR:
[chimera.zip](https://github.com/microsoft/vcpkg/files/4936831/chimera.zip)

It can completely avoid building pcre and hyperscan, but due to the direct use of `CMakeLists.txt` in hyperscan, some macros may not be defined.

**Comment by ZMZ91** [comment]
@JackBoosY, just tried the package you uploaded and got following error during installation
[install-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/4941306/install-x64-osx-dbg-out.log)

**Comment by JackBoosY** [comment]
```
In file included from /usr/local/include/boost/smart_ptr/detail/local_sp_deleter.hpp:20:
/usr/local/include/boost/smart_ptr/detail/local_counted_base.hpp:30:28: error: 'local_counted_base' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Werror,-Wweak-vtables]
class BOOST_SYMBOL_VISIBLE local_counted_base
                           ^
/usr/local/include/boost/smart_ptr/detail/local_counted_base.hpp:92:28: error: 'local_counted_impl' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Werror,-Wweak-vtables]
class BOOST_SYMBOL_VISIBLE local_counted_impl: public local_counted_base
```
Seems doesn't link with boost libraries?

**Comment by JackBoosY** [comment]
@ZMZ91 if you agree with my changes, please commit them to this PR and we can solve the failures.

**Comment by ZMZ91** [comment]
@JackBoosY, just checked in following your changes.

**Comment by JackBoosY** [comment]
Missing `intrin.h`, does this port support unix?

**Comment by ZMZ91** [comment]
I haven't tried on unix. Unix is optional and linux and macos are the first priority.

**Comment by JackBoosY** [comment]
@ZMZ91 Okay, let's support windows first.

File: ports/chimera/CONTROL [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Source: chimera
+Version: 5.2.1
+Homepage: https://github.com/intel/hyperscan
+Description: Chimera is a software regular expression matching engine that is a hybrid of Hyperscan and PCRE. The design goals of Chimera are to fully support PCRE syntax as well as to take advantage of the high performance nature of Hyperscan.
+Build-Depends: boost-array, boost-chrono, boost-config, boost-core, boost-crc, boost-detail, boost-functional, boost-regex, boost-system, boost-thread, boost-type-traits, boost-unordered, boost-utility, boost-dynamic-bitset, boost-random, boost-graph, boost-multi-array, boost-icl, boost-ptr-container, python3, ragel, hyperscan
```
Comment (JackBoosY) [inline ports/chimera/CONTROL:5]:
```suggestion
Build-Depends: boost-array, boost-chrono, boost-config, boost-core, boost-crc, boost-detail, boost-functional, boost-regex, boost-system, boost-thread, boost-type-traits, boost-unordered, boost-utility, boost-dynamic-bitset, boost-random, boost-graph, boost-multi-array, boost-icl, boost-ptr-container, python3, ragel, hyperscan
Supports: windows
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/chimera/CONTROL [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
```
Comment (JackBoosY) [inline ports/chimera/CONTROL:5]:
```suggestion
Build-Depends: python3, ragel, hyperscan
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/chimera/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    REPO intel/hyperscan
+    REF c00683d73916e39f01b0d418f686c8b5c379159c
+    SHA512 3c4d52706901acc9ef4c3d12b0e5b2956f4e6bce13f6828a4ba3b736c05ffacb1d733ef9c226988ca80220584525f9cb6dcfe4914ced6cc34ae6a0a45975afb5
+    HEAD_REF master
+    PATCHES fix-cmakelists.patch
+)
+
+vcpkg_find_acquire_program(PYTHON3)
+get_filename_component(PYTHON3_DIR ${PYTHON3} DIRECTORY)
+set(ENV{PATH} "$ENV{PATH};${PYTHON3_DIR}")
```
Comment (NancyLi1013) [inline ports/chimera/portfile.cmake:15]:
```suggestion
vcpkg_add_to_path(${PYTHON3_DIR})
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
@ZMZ91 It should be noted that `intrin.h` cannot be found in linux and osx, so these two triplets are not currently supported.

**Comment by ZMZ91** [comment]
Hi, just tried on my mac, still got the error 
```
In file included from /usr/local/include/boost/smart_ptr/detail/local_sp_deleter.hpp:20:
/usr/local/include/boost/smart_ptr/detail/local_counted_base.hpp:30:28: error: 'local_counted_base' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Werror,-Wweak-vtables]
class BOOST_SYMBOL_VISIBLE local_counted_base
                           ^
/usr/local/include/boost/smart_ptr/detail/local_counted_base.hpp:92:28: error: 'local_counted_impl' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Werror,-Wweak-vtables]
class BOOST_SYMBOL_VISIBLE local_counted_impl: public local_counted_base
                           ^
In file included from /Users/zmz/github/vcpkg/buildtrees/hyperscan/src/v5.2.1-c4e657f760.clean/tools/hscollider/NfaGeneratedCorpora.cpp:40:
In file included from /Users/zmz/github/vcpkg/buildtrees/hyperscan/src/v5.2.1-c4e657f760.clean/src/nfagraph/ng.h:44:
In file included from /Users/zmz/github/vcpkg/buildtrees/hyperscan/src/v5.2.1-c4e657f760.clean/src/util/graph.h:42:
In file included from /usr/local/include/boost/graph/depth_first_search.hpp:25:
In file included from /usr/local/include/boost/optional.hpp:15:
In file included from /usr/local/include/boost/optional/optional.hpp:32:
/usr/local/include/boost/optional/bad_optional_access.hpp:22:7: error: 'bad_optional_access' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Werror,-Wweak-vtables]
class bad_optional_access : public std::logic_error
      ^
3 errors generated.
```

**Comment by JackBoosY** [comment]
@ZMZ91 We can set it to support Windows first.

**Comment by ZMZ91** [comment]
> @ZMZ91 We can set it to support Windows first.

Sure, but I don't have a windows env but only macos.

**Comment by ZMZ91** [comment]
Thanks for your rapid response @JackBoosY. But I still got same errors using latest code.

**Comment by JackBoosY** [comment]
Fixing x64-osx...

**Comment by JackBoosY** [comment]
@ZMZ91 It seems that chimera is not compatible with macosx at all. I have encountered many syntaxes that clang does not support. So I am going to disable macosx build.

File: ports/chimera/CONTROL [hunk 10]
```diff
@@ -0,0 +1,5 @@
+Source: chimera
+Version: 5.2.1
+Homepage: https://github.com/intel/hyperscan
+Description: Chimera is a software regular expression matching engine that is a hybrid of Hyperscan and PCRE. The design goals of Chimera are to fully support PCRE syntax as well as to take advantage of the high performance nature of Hyperscan.
+Build-Depends: python3, ragel, hyperscan
```
Comment (JackBoosY) [inline ports/chimera/CONTROL:5]:
```suggestion
Build-Depends: python3, ragel, hyperscan
Supports: !osx
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by ZMZ91** [comment]
> @ZMZ91 It seems that chimera is not compatible with macosx at all. I have encountered many syntaxes that clang does not support. So I am going to disable macosx build.

emmm, I don't know why, but my application runs well with chimera on my macbook.

**Comment by JackBoosY** [comment]
For example:
```
/Users/vagrant/Data/buildtrees/chimera/src/b5c379159c-88a7a080a9.clean/chimera/../src/util/hash.h:60:12: error: 'auto' not allowed in function return type
    static auto has_begin_function(const C &obj) -> decltype(std::begin(obj)) {
           ^~~~
```

**Comment by ZMZ91** [comment]
> For example:
> 
> ```
> /Users/vagrant/Data/buildtrees/chimera/src/b5c379159c-88a7a080a9.clean/chimera/../src/util/hash.h:60:12: error: 'auto' not allowed in function return type
>     static auto has_begin_function(const C &obj) -> decltype(std::begin(obj)) {
>            ^~~~
> ```

Is that something about the compiler version? I never got such issue on my macbook. I'm using CMAKE_CXX_STANDARD 17.

**Comment by JackBoosY** [comment]
@ZMZ91 Not working, see https://stackoverflow.com/questions/30665506/auto-not-allowed-in-function-prototype-with-clang

**Comment by ZMZ91** [comment]
> @ZMZ91 Not working, see https://stackoverflow.com/questions/30665506/auto-not-allowed-in-function-prototype-with-clang

Checked on my env, the clang version is
```
Apple clang version 11.0.0 (clang-1100.0.33.16)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
Could you get such an env and try?

**Comment by JackBoosY** [comment]
@ZMZ91 
```
vcpkg@vcpkg % clang --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
vcpkg@vcpkg %
```

**Comment by ZMZ91** [comment]
ha, that's weird. With my previous package, I can install chimera successfully on both mac and linux and my app runs without problem as well.

**Comment by JackBoosY** [comment]
@ZMZ91 Therefore, I think this issue can be left to other contributors.

**Comment by strega-nil** [comment]
I've fixed the problem, hopefully.

**Comment by strega-nil** [comment]
Is there a reason that this isn't just a feature of hyperscan? @ZMZ91

**Comment by ZMZ91** [comment]
> Is there a reason that this isn't just a feature of hyperscan? @ZMZ91

Chimera is indeed a feature from hyperscan. But it requires pcre which makes it quite heavier. Ideally I think it could be built in hyperscan package and add a switch to control chimera feature on/off.

**Comment by strega-nil** [comment]
@ZMZ91 yeah, that's possible; you'd have, in the hyperscan `CONTROL` file:

```
Feature: chimera
Description: Add the chimera regular expression engine
```

and then you can check for the feature in `hyperscan`'s `portfile.cmake` with `"chimera" IN_LIST FEATURES`.

**Review by strega-nil — CHANGES_REQUESTED** [review]
Please make this port a feature of the hyperscan port.

**Comment by NancyLi1013** [comment]
@ZMZ91 
Could you please add `chimera ` as a feature for `hyperscan`?

**Comment by NancyLi1013** [comment]
@ZMZ91 
Could you please handle the conflicts and address the review suggestions?

**Comment by NancyLi1013** [comment]
Pinging @ZMZ91 for response. Is work still being done for this PR?

**Comment by NancyLi1013** [comment]
Thanks for the PR; we're closing this for now since there's been no response. If you'd like to continue working on it, please reopen and ping us!

**Pull request closed without merging**